res/schema/audit-layout.sql

```diff
@@ -43,6 +43,16 @@ CREATE TABLE IF NOT EXISTS message (
 );
 
 
+CREATE TABLE IF NOT EXISTS metadata (
+  id                INTEGER PRIMARY KEY NOT NULL,
+  context           VARCHAR(36)         NOT NULL REFERENCES execution (context),
+  kind              VARCHAR(256)        NOT NULL DEFAULT '',
+  name              VARCHAR(1024)       NOT NULL DEFAULT '',
+  metadata          TEXT                         DEFAULT NULL,
+  UNIQUE (context, kind, name)
+);
```

Comment:
@dkubek  just thinking, could we obtain easily information about the file in which the actor is defined? adding the `path` column (or having it part of metadata?) would be potentially beneficial also in some cases.

Reply:
@fernflower not sure whether I understand it right. what do you mean by function of interest? do you mean the actor definition? that should be present always in the actor.py file. But I dunno whether the framework has a high requirement for that. Speaking about other functions and libraries, we are not interested about them here. We want to know only where the actor is defined.

Reply:
just to add a reason why I am thinking about having the path:
* actor name and directory names differs (and can differ completely)
* we do not have safe namespaces between repositories. it's possible to have 2 actors of the same name on 2 different paths and execute them. in this case, having information about that could be valuable as we will know that there is something fishy

for the experiment, it's possible to `cp -a <existing-actor> <my-existing-actor>`. now, let's imagine someone creates a generic actor name like "read_configuration". Then we are getting to dark waters where we have ambiguous actor names.

Reply:
not saying it's critical to have it, just that it could help us to detect such problems if happens.

Reply:
You are right, I thought it's impossible to have 2 actors with the same name. Then let's add path to actor file.

Reply:
@pirat89 @fernflower Actually, the ``path`` is already present in the metadata field. Sorry, I should have been more explicit in what the metadata looks like. Here is a sample from a random actor:

```json
{
  "class_name": "ScanSystemdSource",
  "path": "/usr/share/leapp-repository/repositories/system_upgrade/common/actors/systemd/scansystemdsource",
  "name": "scan_systemd_source",
  "tags": [
    "IPUWorkflowTag",
    "FactsPhaseTag"
  ],
  "consumes": [],
  "produces": [
    "SystemdServicesInfoSource",
    "SystemdServicesPresetInfoSource",
    "SystemdBrokenSymlinksSource"
  ],
  "dialogs": [],
  "description": "\n    Provides info about systemd on the source system\n\n    The provided info includes information about:\n    - vendor presets of services\n    - systemd service files, including their state\n    - broken systemd symlinks\n\n    There is an analogous actor :class:`ScanSystemdTarget` for target system.\n    ",
  "apis": [],
  "phase": "FactsCollection"
}
```

I don't think having a separate column for ``path`` is necessary:

1. Currently, the ``metadata`` table stores information about different kinds of objects (differentiated by the ``kind`` column). Concretely, I am saving also workflow metadata, which does not have any path associated with it. So the ``path`` column would be only meaningful for some entries, which is something I am not that big of a fan of doing if I can avoid it
2. I don't see a use case right now where I would need to query metadata given an actor path, so it would be redundant in this way. But definitely correct me if I'm wrong and miss something.

However, any application needing the path of an actor (given actor name) would have to parse it from the json, which _is_ an inconvenience, however, we have to parse the metadata as json later anyway, so I am not sure whether it has too much value to have this one specific information separate.

One thing however, I would like your opinion on is whether to make the data that is stored more explicit in the source code. Seeing that you both missed that the ``path`` is already being stored, means I have done a very poor job of telegraphing what data is actually being saved. Do you think it would be better to manually list all the fields I am storing in the metadata column?

Reply:
@dkubek  thank you for detailed explanation. I agree with you that the separate column is not needed in such a case. Having it as a part of metdata is completely ok, especially in case we have already the data there :)

Reply:
Can you put info about what is now stored inside the metadata as part of the commit msg? I think that listing the keys should be ok.